### PR TITLE
Make global JVM arguments multi-line

### PR DESCRIPTION
**Pull request by ryanccn**
The individual instance JVM arguments option already uses a multiline text edit. Because there's a lot of text involved in 
there, multi-line might be more suitable.

### CODE REVIEW

**Comment by Scrumplex** [comment]
you based this branch off your trash-instances branch

**Comment by Scrumplex** [comment]
oh you already fixed, nvm

**Comment by ryanccn** [comment]
Also, newlines are rendered as spaces in `QLineEdit` (at least on my end), so it could cause some confusion when options are copy-pasted into the text box.

**Review by timoreo22 — APPROVED** [review]

**Review by flowln — APPROVED** [review]
Why two merges? lol :p

File: launcher/ui/pages/global/JavaPage.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ void JavaPage::applySettings()
 
     // Java Settings
     s->set("JavaPath", ui->javaPathTextBox->text());
-    s->set("JvmArgs", ui->jvmArgsTextBox->text());
+    s->set("JvmArgs", ui->jvmArgsTextBox->toPlainText());
```
Comment (kthchew) [inline launcher/ui/pages/global/JavaPage.cpp]:
```suggestion
    s->set("JvmArgs", ui->jvmArgsTextBox->toPlainText().replace("\n", " "));
```

The behavior when a user saves arguments with new lines differs between the global setting and the instance setting. This wasn't a problem before because the old global arguments text box didn't accept new lines.

In the instance setting (already multi-line), when a user enters newlines into the JVM arguments box and saves their settings, the new lines are automatically converted to spaces and saved. For reference, it does this:

https://github.com/PolyMC/PolyMC/blob/699ad316f0d90580fa13d570d6c25aff903a470d/launcher/ui/pages/instance/InstanceSettingsPage.cpp#L195

In this new multi-line global settings text box, if the user types new lines, the new lines are saved. If a user saves arguments with new lines, the game crashes on startup because it doesn't recognize that the new lines separate different arguments (at least on my system):

```
Unrecognized VM option 'UseG1GC
-Dsun.rmi.dgc.server.gcInterval=2147483646
-XX:+UnlockExperimentalVMOptions
-XX:G1NewSizePercent=20
-XX:G1ReservePercent=20
-XX:MaxGCPauseMillis=50
-XX:G1HeapRegionSize=32M'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Minecraft process ID: 86444


Process exited with code 1.
```

tl;dr: the behavior when users enter a new line into the arguments box should be standard between the global and instance settings. The easiest way to do this would just be to convert new lines to spaces here.

**Review by kthchew — CHANGES_REQUESTED** [review]

File: launcher/ui/pages/global/JavaPage.cpp [hunk 2]
```diff
@@ -166,7 +166,7 @@ void JavaPage::on_javaTestBtn_clicked()
         return;
     }
     checker.reset(new JavaCommon::TestCheck(
-        this, ui->javaPathTextBox->text(), ui->jvmArgsTextBox->text(),
+        this, ui->javaPathTextBox->text(), ui->jvmArgsTextBox->toPlainText(),
```
Comment (Scrumplex) [inline launcher/ui/pages/global/JavaPage.cpp]:
```suggestion
        this, ui->javaPathTextBox->text(), ui->jvmArgsTextBox->toPlainText().replace("\n", " "),
```

**Review by Scrumplex — APPROVED** [review]

**Review by txtsd — APPROVED** [review]
